HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentComponentsController.swift

```diff
@@ -48,24 +72,43 @@ public final class PaymentComponentsController: NSObject {
         }
     }
 
-    var isLoading: Bool = false {
+    /// reponsible for storing the loading state of the controller and passing it to the delegate listeners
+    private var isLoading: Bool = false {
         didSet {
             delegate?.isLoadingStateChanged(isLoading: isLoading)
         }
     }
 
-    var paymentComponentView: PaymentComponentView!
-
+    /**
+     Returing the first payment provider installed if is available and installed
+     - Returns: a Payment Provider object.
+     */
     public func obtainFirstPaymentProvider() -> PaymentProvider? {
         paymentProviders.first
     }
 
+    /**
+     Checks if the document is payable which looks for iban extraction.
```

Comment:
[suggestion]: I will rewrite this like this:
```
This function checks if the document is payable by extracting the IBAN.

    - Parameters:
        - docId: The ID of the uploaded document.
        - completion: A closure for processing asynchronous data received from the service. It has a Result type parameter, representing either success or failure. The completion block is called on the main thread. 
        In the case of success, it includes a boolean value indicating whether the IBAN was extracted successfully. 
        In case of failure, it returns an error from the server side.
```